be2-scala/src/main/scala/ch/epfl/pop/jsonNew/HighLevelProtocol.scala

```diff
@@ -0,0 +1,84 @@
+package ch.epfl.pop.jsonNew
+
+import ch.epfl.pop.model.network.{ErrorObject, JsonRpcRequest, JsonRpcResponse, MethodType}
+import ch.epfl.pop.model.network.MethodType.MethodType
+import ch.epfl.pop.model.network.method._
+import ch.epfl.pop.model.network.method.message.Message
+import ch.epfl.pop.model.objects.{Base64Data, Hash, PublicKey, Signature, WitnessSignaturePair}
+import ObjectProtocol._
+import spray.json._
+
+object HighLevelProtocol extends DefaultJsonProtocol {
+
+  // ----------------------------------- ENUM FORMATTERS ----------------------------------- //
+  implicit object methodTypeFormat extends RootJsonFormat[MethodType] {
+    override def read(json: JsValue): MethodType = json match {
+      case JsString(method) => MethodType.unapply(method).getOrElse(MethodType.INVALID)
+      case _ => throw new IllegalArgumentException(s"Can't parse json value $json to a MethodType")
+    }
+
+    override def write(obj: MethodType): JsValue = JsString(obj.toString)
+  }
+
+
+  // ----------------------------------- HIGH-LEVEL FORMAT ----------------------------------- //
+  implicit object messageFormat extends RootJsonFormat[Message] {
+    final private val PARAM_DATA: String = "data"
+    final private val PARAM_SENDER: String = "sender"
+    final private val PARAM_SIGNATURE: String = "signature"
+    final private val PARAM_MESSAGE_ID: String = "message_id"
+    final private val PARAM_WITNESS_SIG: String = "witness_signatures"
+
+    override def read(json: JsValue): Message =
+      json.asJsObject.getFields(PARAM_DATA, PARAM_SENDER, PARAM_SIGNATURE, PARAM_MESSAGE_ID, PARAM_WITNESS_SIG) match {
+        case Seq(data@JsString(_), sender@JsString(_), signature@JsString(_), messageId@JsString(_), JsArray(witnessSig)) =>
+          Message(
+            data.convertTo[Base64Data],
+            sender.convertTo[PublicKey],
+            signature.convertTo[Signature],
+            messageId.convertTo[Hash],
+            witnessSig.map(_.convertTo[WitnessSignaturePair]).toList
+          )
+        case _ => throw new IllegalArgumentException(s"Can't parse json value $json to a Message object")
+      }
+
+    override def write(obj: Message): JsValue =
+      JsObject(
+        PARAM_DATA -> obj.data.toJson,
+        PARAM_SENDER -> obj.sender.toJson,
+        PARAM_SIGNATURE -> obj.signature.toJson,
+        PARAM_MESSAGE_ID -> obj.message_id.toJson,
+        PARAM_WITNESS_SIG -> obj.witness_signatures.toJson // FIXME does this work? Otherwise use "JsArray(obj.witness_signatures.map(_.toJson).toVector))"
```

Comment:
Yes....... wheeeen the code will compile ^^'. Will answer other messages tomorrow, but in short, since the graph doesn't have a global reference to dbActorNew yet, the compilation fails. This is an easy fix but I'm not quite convinced we have the best approach for it yet